### ISSUE TITLE
Removal of Summarizable and toArray

### ISSUE THREAD

**Issue by Sibyx** [Sibyx/phpGPX#69]
As part of the new major release, I propose the removal of the `Summarizable` interface along with its `toArray` method. This decision stems from my belief that all models should instead implement the `JsonSerializable` interface, thereby returning a [GeoJSON](https://datatracker.ietf.org/doc/html/rfc7946) compatible result where feasible.

The usage of arrays in this context seems antiquated and unnecessarily complex, hence it doesn't align with the direction of our project. I anticipate that the removal of these features won't be a significant loss for the users. However, I welcome any thoughts and potential concerns about this proposed change.

_Originally posted by @Sibyx in https://github.com/Sibyx/phpGPX/discussions/67#discussioncomment-6644721_

**Comment by miqwit** [comment]
Exporting in GeoJSON is a must. This library could then act as a GPX to GeoJSON converter (like some others - not much - but with the other features like the stats that make this library charming).

**Comment by Sibyx** [comment]
Yep, totally agree. There is already some implementation started in `develop`.

**Comment by Sibyx** [comment]
Implemented and will be released with `2.x`
